scanner/src/main/kotlin/scanners/fossid/FossId.kt

```diff
@@ -332,52 +334,57 @@ class FossId(
+                        checkAndCreateScan(scans, revision, url, path, projectCode, projectName)
+                    }
 
-                val provenance = RepositoryProvenance(pkg.vcsProcessed, pkg.vcsProcessed.revision)
+                    val provenance = RepositoryProvenance(pkg.vcsProcessed, pkg.vcsProcessed.revision)
 
-                if (waitForResult) {
-                    val rawResults = getRawResults(scanCode)
-                    val resultsSummary = createResultSummary(startTime, provenance, rawResults)
+                    if (waitForResult) {
+                        val rawResults = getRawResults(scanCode)
+                        val resultsSummary = createResultSummary(startTime, provenance, rawResults)
 
-                    results.getOrPut(pkg) { mutableListOf() } += resultsSummary
-                } else {
-                    val issue = OrtIssue(
-                        source = pkg.id.toCoordinates(),
-                        message = "This package has been scanned in asynchronous mode. Scan results are available " +
-                                "on the FossID instance.",
-                        severity = Severity.HINT
-                    )
-                    val summary = ScanSummary(
-                        startTime = startTime,
-                        endTime = Instant.now(),
-                        packageVerificationCode = "",
-                        licenseFindings = sortedSetOf(),
-                        copyrightFindings = sortedSetOf(),
-                        issues = listOf(
-                            issue
+                        results.getOrPut(pkg) { mutableListOf() } += resultsSummary
+                    } else {
+                        val issue = OrtIssue(
+                            source = pkg.id.toCoordinates(),
+                            message = "This package has been scanned in asynchronous mode. Scan results are " +
+                                    "available on the FossID instance.",
+                            severity = Severity.HINT
+                        )
+                        val summary = ScanSummary(
+                            startTime = startTime,
+                            endTime = Instant.now(),
+                            packageVerificationCode = "",
+                            licenseFindings = sortedSetOf(),
+                            copyrightFindings = sortedSetOf(),
+                            issues = listOf(
+                                issue
+                            )
                         )
-                    )
 
-                    val scanResult = ScanResult(provenance, details, summary)
-                    results.getOrPut(pkg) { mutableListOf() } += scanResult
+                        val scanResult = ScanResult(provenance, details, summary)
+                        results.getOrPut(pkg) { mutableListOf() } += scanResult
+                    }
+                } catch (e: IllegalStateException) {
```

Comment:
Is it possible (and does it make sense) to not only log the exception, but create a result with an issue for the current package?